### PR TITLE
Support Chainer-style initialization in TransformDataset

### PR DESCRIPTION
**Pull request by Hakuyume**

### CODE REVIEW

File: chainercv/chainer_experimental/datasets/sliceable/transform_dataset.py [hunk 1]
```diff
@@ -22,20 +24,35 @@ class TransformDataset(GetterDataset):
             This dataset should have :meth:`__len__` and :meth:`__getitem__`.
         keys (int or string or tuple of strings): The number or name(s) of
             data that the transform function returns.
+            If this parametr is omitted, :meth:`__init__` fetches a sample
```
Comment (yuyu2172) [inline chainercv/chainer_experimental/datasets/sliceable/transform_dataset.py]: parameter

**Comment by yuyu2172** [comment]
pfnCI, test this please

**Comment by Hakuyume** [comment]
pfnCI, test this please

**Comment by Hakuyume** [comment]
pfnCI, test this please

**Comment by Hakuyume** [comment]
pfnCI, test this please

**Pull request merged**